### ISSUE TITLE
Haplotype::copy unsorted or overlapping regions provided

### ISSUE THREAD

**Issue by bredelings** [luntergroup/octopus#121]
**Describe the bug**

While running octopus (current develop branch) I see the the following error output, and the run then hangs:
```
[2020-05-15 00:25:44] <INFO> NODE_823_length_10242_cov_14.583669:3363          73.3%          17m 27s            6m 22s
[2020-05-15 00:26:01] <INFO> NODE_845_length_10011_cov_40.083551:9574          74.3%          17m 44s             6m 9s
[2020-05-15 00:26:10] <EROR> Encountered a problem whilst calling NODE_862_length_9817_cov_44.884908:0-9817(Haplotype::copy unsorted or overlapping regions provided)
[2020-05-15 00:27:41] <INFO> NODE_849_length_9968_cov_31.560712:3907           75.3%          19m 24s            6m 23s
```
octopus keeps on running, but nothing more is written.

**Version**

```shell
$ octopus --version
octopus version 0.7.0 (develop ba99b500)
Target: x86_64 Linux 2.6.32-642.13.1.el6.x86_64
SIMD extension: SSE2
Compiler: GNU 9.1.1
Boost: 1_73
```

Command line to run octopus:
```shell
$ octopus -R contigs-1k+.fasta -I contigs-1k+.bam -o contigs-1k+.g.vcf.gz --threads 32 --refcall POSITIONAL
```

[NODE862.tar.gz](https://github.com/luntergroup/octopus/files/4632268/NODE862.tar.gz)

**Comment by dancooke** [comment]
Thanks for the bug report. Should be fixed in 2bde043330a2f2c36fcca958e7fa6887653d88ab.

**Comment by bredelings** [comment]
Thanks!
